app/src/main/java/com/github/mateo762/myapplication/home/fragments/SummaryFragment.kt

```diff
@@ -1,36 +1,128 @@
 package com.github.mateo762.myapplication.home.fragments
 
 import android.content.ContentValues.TAG
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
 import android.os.Build
 import android.os.Bundle
 import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.annotation.RequiresApi
 import androidx.compose.foundation.layout.*
 import androidx.compose.material.*
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.ui.platform.ComposeView
 import androidx.fragment.app.Fragment
 import com.github.mateo762.myapplication.getHardCodedHabits
+import com.github.mateo762.myapplication.models.HabitEntity
+import com.github.mateo762.myapplication.room.HabitRepository
 import com.github.mateo762.myapplication.ui.home.HabitListScreen
+import com.github.mateo762.myapplication.ui.home.TodayScreen
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.*
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.time.LocalDateTime
+import javax.inject.Inject
 
-
+@AndroidEntryPoint
 class SummaryFragment : Fragment() {
+
+    private lateinit var habitsRef: DatabaseReference
+    private val habitsState = mutableStateOf(emptyList<HabitEntity>())
+    private val TAG = TodayFragment::class.java.simpleName
+
+    @Inject
+    lateinit var habitRepository: HabitRepository
+
     @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
     ): View {
-        try {
-            return ComposeView(requireContext()).apply {
-                setContent {
-                    HabitListScreen(habits = getHardCodedHabits())
+        return ComposeView(requireContext()).apply {
+            setContent {
+                HabitListScreen(
+                    habits = habitsState.value
+                )
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        // Get the connectivityManager to verify network functions
+        val connectivityManager =
+            context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val activeNetwork = connectivityManager.activeNetwork
+
+        // Verify we have connection -- this way we will at least always run the Listener,
+        // and if Firebase fails, then we run the failed action onCancelled
+        val networkCapabilities = connectivityManager.getNetworkCapabilities(activeNetwork)
+
+        val connectionExists =
+            networkCapabilities?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false
+
+        // Connect to Firebase for real time data
+        if (connectionExists) {
```

Comment:
Just so that I understand, is this check to allow different behavior for the offline mode? Otherwise, why would you need to check the existence of connection?

Reply:
I believe this check allows the OnCancelled function to work correctly. So if for instance, the device isn't connected to wifi, then it will be possible to detect this and load from cache. Received help from Jorge on this, will follow-up with him to confirm ;)